x/evm/keeper/params.go

```diff
@@ -2,57 +2,114 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
 	"github.com/evmos/ethermint/x/evm/types"
+	gogotypes "github.com/gogo/protobuf/types"
 )
 
 // GetParams returns the total set of evm parameters.
 func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
-	// TODO: update once https://github.com/cosmos/cosmos-sdk/pull/12615 is merged
-	// and released
-	for _, pair := range params.ParamSetPairs() {
-		k.paramSpace.GetIfExists(ctx, pair.Key, pair.Value)
+	evmDenom := k.GetEVMDenom(ctx)
+	enableCall := k.GetEnableCall(ctx)
+	enableCreate := k.GetEnableCreate(ctx)
+	chainCfg := k.GetChainConfig(ctx)
+	extraEIPs := k.GetExtraEIPs(ctx)
+	allowUnprotectedTx := k.GetAllowUnprotectedTxs(ctx)
+
+	return types.NewParams(evmDenom, allowUnprotectedTx, enableCreate, enableCall, chainCfg, extraEIPs)
+}
+
+// SetParams Sets the EVM params each in its individual key for better get performance
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	if err := params.Validate(); err != nil {
+		return err
 	}
+	store := ctx.KVStore(k.storeKey)
+
+	chainCfgBz := k.cdc.MustMarshal(&params.ChainConfig)
+	extraEIPsBz := k.cdc.MustMarshal(&params.ExtraEips)
+	evmDenomBz := k.cdc.MustMarshal(&gogotypes.StringValue{Value: params.EvmDenom})
+	allowUnprotectedTxsBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.AllowUnprotectedTxs})
+	enableCallBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.EnableCall})
+	enableCreateBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.EnableCreate})
```

Comment:
ditto, use `[]byte(0x01)` (`true`). For `false` you don't need to set the value and you can use `store.Has`